### PR TITLE
[cu-33va88k] Remove fork and add pallet-proxy with a wrapper to Dali and Picasso

### PR DESCRIPTION
**Pull request by vimukthi-git**
Remove fork and add pallet-proxy with a wrapper to Dali and Picasso

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-33va88k Add proxy pallet to Picasso ](https://app.clickup.com/t/33va88k)

File: code/parachain/frame/composable-traits/src/account_proxy.rs [hunk 1]
```diff
@@ -104,3 +113,27 @@ pub trait AccountProxy {
 		force_proxy_type: Option<Self::ProxyType>,
 	) -> Result<ProxyDefinition<Self::AccountId, Self::ProxyType, Self::BlockNumber>, DispatchError>;
 }
+
+
+pub struct AccountProxyWrapper<Runtime> {
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/account_proxy.rs:118]: Main change

File: code/parachain/runtime/picasso/src/lib.rs [hunk 2]
```diff
@@ -646,6 +646,29 @@ parameter_types! {
 	pub const LockCrowdloanRewards: bool = true;
 }
 
+parameter_types! {
```
Comment (vimukthi-git) [inline code/parachain/runtime/picasso/src/lib.rs:649]: Main change

File: code/parachain/runtime/dali/src/lib.rs [hunk 3]
```diff
@@ -689,21 +690,22 @@ parameter_types! {
...
 	type MaxProxies = MaxProxies;
-	type WeightInfo = weights::account_proxy::WeightInfo<Runtime>;
+	type WeightInfo = weights::proxy::WeightInfo<Runtime>;
 	type MaxPending = MaxPending;
 	type CallHasher = BlakeTwo256;
 	type AnnouncementDepositBase = ProxyPrice;
 	type AnnouncementDepositFactor = ProxyPrice;
 }
 
+type AccountProxyWrapperInstance = AccountProxyWrapper<Runtime>;
```
Comment (vimukthi-git) [inline code/parachain/runtime/dali/src/lib.rs:708]: Main change

**Comment by vimukthi-git** [comment]
To reviewers: Not yet tested with Picasso, I plan to do soon.

File: code/parachain/frame/composable-traits/src/account_proxy.rs [hunk 4]
```diff
@@ -104,3 +113,27 @@ pub trait AccountProxy {
...
+	fn add_proxy_delegate(delegator: &Self::AccountId, delegatee: Self::AccountId, proxy_type: Self::ProxyType, delay: Self::BlockNumber) -> DispatchResult {
+		Self::Proxy::add_proxy_delegate(delegator, delegatee, proxy_type, delay)
+	}
+
+	fn remove_proxy_delegate(delegator: &Self::AccountId, delegatee: Self::AccountId, proxy_type: Self::ProxyType, delay: Self::BlockNumber) -> DispatchResult {
+		Self::Proxy::remove_proxy_delegate(delegator, delegatee, proxy_type, delay)
+	}
+
+	fn find_proxy(real: &Self::AccountId, delegate: &Self::AccountId, force_proxy_type: Option<Self::ProxyType>) -> Result<ProxyDefinition<Self::AccountId, Self::ProxyType, Self::BlockNumber>, DispatchError> {
+		Self::Proxy::find_proxy(real, delegate, force_proxy_type).map(|proxy| ProxyDefinition::from(proxy))
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/account_proxy.rs:137]:
```suggestion
		Self::Proxy::find_proxy(real, delegate, force_proxy_type).map(|proxy| proxy.into())
```

File: code/parachain/runtime/picasso/src/weights/proxy.rs [hunk 5]
```diff
@@ -0,0 +1,96 @@
+// This file is part of Substrate.
```
Comment (vimukthi-git) [inline code/parachain/runtime/picasso/src/weights/proxy.rs:1]: Needs to do benchmarking once in release branch again

**Review by PoisonPhang — APPROVED** [review]

File: code/parachain/frame/composable-traits/src/account_proxy.rs (hunk 1 above)
Comment (vimukthi-git) [reply code/parachain/frame/composable-traits/src/account_proxy.rs:118]: Add comment on the purpose

**Review by andyjsbell — APPROVED** [review]
Thanks @vimukthi-git

**Review by KaiserKarel — COMMENTED** [review]
Why do we need the ProxyWrapper?

**Comment by vimukthi-git** [comment]
@mergifyio refresh

**Review by KaiserKarel — APPROVED** [review]

**Pull request merged**